dali/python/nvidia/dali/tensors.py

```diff
@@ -18,3 +18,86 @@
 from nvidia.dali.backend import TensorListCPU
 from nvidia.dali.backend import TensorGPU
 from nvidia.dali.backend import TensorListGPU
+
+
+def _transfer_to_cpu(data, device):
+    if device.lower() == 'gpu':
+        return data.as_cpu()
+    return data
+
+
+def _join_string(data, crop, edgeitems, sep=', '):
+    if crop:
+        data = data[:edgeitems] + ['...'] + data[-edgeitems:]
+
+    return sep.join(data)
+
+
+def _tensor_to_string(self):
+    """ Returns string representation of Tensor."""
+    import numpy as np
+
+    type_name = type(self).__name__
+    indent = ' ' * 4
+    layout = self.layout()
+    data = np.array(_transfer_to_cpu(self, type_name[-3:]))
+    data_str = np.array2string(data, prefix=indent, edgeitems=2)
+
+    params = [f'{type_name}(\n{indent}{data_str}', f'dtype={self.dtype}'] + \
+        ([f'layout={layout}'] if layout else []) + \
+        [f'shape={self.shape()})']
+
+    return _join_string(params, False, 0, ',\n' + indent)
+
+
+def _tensorlist_to_string(self, indent=''):
+    """ Returns string representation of TensorList."""
+    import numpy as np
+
+    edgeitems = 2
+    spaces_indent = indent + ' ' * 4
+    type_name = type(self).__name__
+    layout = self.layout()
+    shape = list(map(str, self.shape()))
+    data = _transfer_to_cpu(self, type_name[-3:])
+    data_str = '[]'
+    crop = False
+
+    if data:
+        if data.is_dense_tensor():
+            data_str = np.array2string(np.array(data.as_tensor()),
+                                       prefix=spaces_indent, edgeitems=edgeitems)
+        else:
+            data = list(data)
+
+            # If number of tensors > 5 returns 2 first and 2 last tensors seperated by dots.
+            crop = len(data) > edgeitems * 2 + 1
```

Comment:
Maybe it can be configurable - like https://numpy.org/doc/stable/reference/generated/numpy.set_printoptions.html ?

Reply:
Or depend on the size of each element. Printing a 10 [1x1] tensor should be not a big deal.

Reply:
This threshold for number of elements needed to start summarization seems like a very good idea.

I could also make the whole thing configurable similarly to [numpy.set_printoptions](https://numpy.org/doc/stable/reference/generated/numpy.set_printoptions.html) allowing user to specify parameters that [numpy.array2string](https://numpy.org/doc/stable/reference/generated/numpy.array2string.html) takes, but I'm not sure if anyone would use it?

Reply:
Just an idea.